src/components/Table/utils/ColumnStorage.ts

```diff
@@ -0,0 +1,42 @@
+import { autorun, reaction } from "mobx";
+import { ColumnStates } from "src/components/Table/utils/ColumnStates";
+import { loadArrayOrUndefined } from "src/components/Table/utils/utils";
+
+/** Loads/saves the column state from sessionStorage. */
+export class ColumnStorage {
+  private expandedIds: string[] | undefined;
+  private visibleIds: string[] | undefined;
+
+  constructor(private states: ColumnStates) {}
+
+  loadExpanded(persistCollapse: string): void {
+    const key = `expandedColumn_${persistCollapse}`;
+    this.expandedIds = loadArrayOrUndefined(key);
+    reaction(
+      () => this.states.expandedColumns.map((cs) => cs.column.id),
+      (columnIds) => sessionStorage.setItem(key, JSON.stringify(columnIds)),
+    );
+  }
+
+  loadVisible(storageKey: string): void {
+    this.visibleIds = loadArrayOrUndefined(storageKey);
+    // Unlike the others, where we only store the value on change, we immediately
+    // store this value (but I'm not sure why...), hence using `autorun`.
+    autorun(() => {
+      const columnIds = this.states.allVisibleColumns.map((cs) => cs.column.id);
+      sessionStorage.setItem(storageKey, JSON.stringify(columnIds));
+    });
```

Comment:
**question:** Are you saying we don't update the storage value whenever the user changes its visible columns?

Reply:
Ah no, we do update it after every change, it's just that we also update it _immediately_, with whatever the default value is. Our other "persist" stuff didn't bother storing the default state, until the user changed something. Not sure that it matters, it's just different between the two, hence making the in-source comment.